File: web/src/types/RendererState.ts
import type MarkdownNode from "./MarkdownNode";
import type Renderer from "./Renderer";

export default interface RendererState {
	renderersMap: Map<string, Renderer>;

	output: string;
	footnotes: MarkdownNode[];
	footnoteRefs: Record<string, MarkdownNode>;

	listDepth: number;
	lineWidth?: number;
}

File: zig/src/types/RendererState.zig
const std = @import("std");

pub const RendererState = struct {
    allocator: std.mem.Allocator,
    renderersMap: std.StringHashMap(*const Renderer),

    output: std.ArrayList(u8),
    footnotes: std.ArrayList(*const MarkdownNode),
    footnoteRefs: std.StringHashMap(*const MarkdownNode),

    listDepth: usize = 0,
    line_width: ?usize = null,
};

const Renderer = @import("Renderer.zig").Renderer;
const MarkdownNode = @import("MarkdownNode.zig").MarkdownNode;
